modules/cloud-router-connection/main.tf

```diff
@@ -0,0 +1,179 @@
+terraform {
+  required_providers {
+    equinix = {
+      source  = "equinix/equinix"
+      version = ""
+    }
+  }
+}
+data "equinix_fabric_service_profiles" "zside" {
+    count = var.zside_ap_type == "SP" ? 1 : 0
+  filter {
+    property = "/name"
+    operator = "="
+    values   = [var.zside_fabric_sp_name]
+  }
+}
+data "equinix_fabric_ports" "zside" {
+    count  = var.zside_ap_type == "COLO" ? 1 : 0
+  filters {
+    name   = var.zside_port_name
+  }
+}
+
+
+resource "equinix_fabric_connection" "primary_cloud_router_connection" {
+  name = var.connection_name
+  type = var.connection_type
+  notifications {
+    type   = var.notifications_type
+    emails = var.notifications_emails
+  }
+  additional_info = var.additional_info != [] ? var.additional_info : null
+  bandwidth       = var.bandwidth
+  redundancy { priority = "PRIMARY" }
+  order {
+    purchase_order_number = var.purchase_order_number
+  }
+  a_side {
+    access_point {
+      type = var.aside_ap_type
+      router {
+        uuid = var.aside_fcr_uuid
+      }
+    }
+  }
+  dynamic "z_side" {
+    #ports z_side type
+    for_each = var.zside_ap_type == "COLO" ? [1] : []
+    content {
+      access_point {
+          type = var.zside_ap_type
+        port {
+          uuid = data.equinix_fabric_ports.zside[0].id
+        }
+        link_protocol {
+          type       = one(data.equinix_fabric_ports.zside[0].data[0].encapsulation).type
+          vlan_tag   = one(data.equinix_fabric_ports.zside[0].data[0].encapsulation).type == "DOT1Q" ? var.zside_vlan_tag : null
+          vlan_c_tag = one(data.equinix_fabric_ports.zside[0].data[0].encapsulation).type == "QINQ" ? var.zside_vlan_tag : null
```

Comment:
This is reversed. If it's a QINQ Connection with just one tag the stag will be the outer tag. Better to make the second variable be `zside_vlan_inner_tag` and have the `zside_vlan_tag` be assigned to stag and the `zside_vlan_inner_tag` be assigned to the ctag